connpolicy_test.go

```diff
@@ -0,0 +1,177 @@
+)
+
+func init() {
+	rootCertPool = x509.NewCertPool()
+	caBuff, _ := base64.StdEncoding.DecodeString(rootCA)
+	cert, _ := x509.ParseCertificate(caBuff)
+
+	rootCertPool.AddCert(cert)
+
+	caddy.StartAdmin([]byte(`{}`))
+}
+
+func loadConf(input []byte) error {
+	buf := bytes.NewBuffer(input)
+	fmt.Println("config to load: ", buf.String())
+
+	resp, err := http.Post("http://localhost:2019/load", "application/json", buf)
+	if err != nil {
+		return err
+	}
+
+	if code := resp.StatusCode; code != 200 {
+		return fmt.Errorf("response is not 200 but :%d", code)
+	}
+
+	return nil
+}
+
+func buildTestAuthTLSConf(t, v string) []byte {
+	start, end := []byte(`{
+	"apps": {
+		"http": {
+			"http_port": 8080,
+			"https_port": 4430,
+			"grace_period": "2s",
+			"servers": {
+				"mainServices": {
+					"listen": [
+						":4430"
+					],
+					`),
+		[]byte(`"routes": [
+						{
+							"match": [
+								{
+									"host": [
+										"localhost"
+									]
+								}
+							],
+							"handle": [
+								{
+									"handler": "static_response",
+									"status_code": 204,
+									"headers": {
+										"Given-Host": [
+											"{http.request.host}"
+										]
+									},
+									"close": true
+								}
+							]
+						}
+					]
+				}
+			}
+		}
+	},
+	"tls": {
+		"certificates": {
+			"load_files": [
+				{
+					"certificate": "./server.crt",
+					"key": "./server.key",
+					"format": "pem",
+					"tags": []
+				}
+			]
+		}
+	}
+}`)
```

Comment:
None of this is necessary either.

Just make a TLSConnPolicy that has the values you want to test.

Going through all these other parts of the software is bad form in unit testing.